app/services/register_and_partner_api/sync_users.rb

```diff
@@ -54,41 +54,39 @@ def self.user_attributes_from(user_from_api)
       }
     end
 
-    def self.determine_core_induction_programme(cip_name)
-      case cip_name
-      when "ambition"
-        CIP_TYPES[:ambition]
-      when "edt"
-        CIP_TYPES[:edt]
-      when "teach_first"
-        CIP_TYPES[:teach_first]
-      when "ucl"
-        CIP_TYPES[:ucl]
-      end
-    end
-
     def self.find_or_create_user(attributes, profile_class)
       user = ::User.find_or_initialize_by(register_and_partner_id: attributes[:register_and_partner_id])
-      profile = profile_class.find_or_initialize_by(user: user)
+      needs_inviting = !user.persisted?
```

Comment:
negation sozzle my head at times. what's !user checking with this persisted?

Reply:
`user.persisted?` checks whether the user was already saved in the database. `!user.persisted?` is a negation of that, so it checks whether the user was never saved - so whether they are a new user.